### PR TITLE
Throttle optimizer monitor

### PR DESCRIPTION
**Pull request by mvpatel2000**
# What does this PR do?

The optimizer monitor logs very aggressively. This PR changes default to log every 10 batches. It also cleans up comments and removes some logging which is not useful per @bmosaicml's recommendation

### CODE REVIEW

File: composer/optim/decoupled_weight_decay.py [hunk 1]
```diff
@@ -193,18 +193,10 @@ class DecoupledAdamW(AdamW):
             lambda param, optim_state, step_tensor: torch.linalg.vector_norm(optim_state['exp_avg']),
         'l2_norm/param':
             lambda param, optim_state, step_tensor: torch.linalg.vector_norm(param.data),
-        'l2_norm/second_moment_sqrt':
```
Comment (j316chuck) [inline composer/optim/decoupled_weight_decay.py:196]: Is this change part of throttling optimizer monitor or should it be a separate pr that changes weight decay logic?

**Review by j316chuck — COMMENTED** [review]
lgtm overall; left a few questions on parts of the PR.

File: tests/callbacks/test_optimizer_monitor.py [hunk 2]
```diff
@@ -151,7 +151,7 @@ def test_fsdp_optimizer_monitor_transformer(device, world_size, tiny_gpt2_model,
                       loggers=in_memory_logger,
                       train_dataloader=train_dataloader,
                       optimizers=DecoupledAdamW(model.parameters()),
-                      max_duration='3ba',
+                      max_duration='11ba',
```
[the remark this answers is not in the document]
Comment (dakinggg) [reply tests/callbacks/test_optimizer_monitor.py:154]: since the default was changed to 10, gotta set max duration beyond 10 to log anything (i presume)

**Review by dakinggg — APPROVED** [review]
LGTM, would like @bmosaicml to approve as well since I have no context for the removal of some metrics

File: tests/callbacks/test_optimizer_monitor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (mvpatel2000) [reply tests/callbacks/test_optimizer_monitor.py:148]: Exactly, I need to extend max dur to extend beyond checkpoint frequency to verify it actually logs at that frequency :)

File: composer/optim/decoupled_weight_decay.py (hunk 1 above)
Comment (mvpatel2000) [reply composer/optim/decoupled_weight_decay.py:196]: This removes logging for some optimizer monitor features that are no longer necessary per @bmosaicml's feedback. It shouldn't change any functional logic, just remove logging :)
